src/components/FlowRunCreateFormV2.vue

```diff
@@ -127,8 +125,7 @@
   const workQueueName = ref<string | null>(props.deployment.workQueueName)
   const retries = ref<number | null>(null)
   const retryDelay = ref<number | null>(null)
-  const jobVariables = ref<string>()
-  const { state: jobVariablesState, error: jobVariablesError } = useValidation(jobVariables, isJson('Job variables'))
+  const jobVariables = ref<string | undefined>(can.access.flowRunInfraOverrides ? '{}' : undefined)
```

Comment:
This change is to make the input here initially have `{}` rather than be empty, _but_ it needs to still be undefined if the flag is off so that the field is dropped when it goes over the wire.

<img width="1151" alt="image" src="https://github.com/PrefectHQ/prefect-ui-library/assets/22418768/fbd0c404-cde8-4efd-9275-3f666c82dcb7">